Sources/Logger/Loggers/FileLogger/FileLogger.swift

```diff
@@ -18,20 +19,41 @@ public class FileLogger: Logging {
 
     public var levels: [Level] = [.info]
 
-    /// FileLogger initializer
+    /// FileLogger initialization
     ///
     /// - Parameters:
-    ///   - subsystem: suit name of the application. Must be passed to create logs from app extensions.
+    ///   - id: Identifier required by `Logging` protocol
+    ///   - userDefaults: `UserDefaults` storage instance, where necessary metadata for the logger are stored.
+    ///   Use different `UserDefaults.init(suiteName:)` when registering multiple instances of `FileLogger`.
+    ///   - fileManager: `FileManager` instance used for persisting logging files
+    ///   - suiteName: name of the application. Must be used if logs are to be shared between the app and its extensions.
+    ///   - logDirectoryName: name of the directory, where logging files are to be stored
+    ///   - fileHeaderContent: such `String` will be placed at the header of each logging file. Empty `String` is used as default.
     ///   - numberOfLogFiles: a number of log files that can be used for logging
-    public init(suiteName: String? = nil, numberOfLogFiles: Int = 4) throws {
-        fileLoggerManager = try FileLoggerManager(
+    public init(
+      id: UUID = UUID(),
+      userDefaults: UserDefaults = UserDefaults.standard,
+      fileManager: FileManager = FileManager.default,
+      suiteName: String? = nil,
+      logDirectoryName: String = "logs",
+      fileHeaderContent: String = "",
+      numberOfLogFiles: Int = 4
+    ) throws {
+        self.id = id
+        self.fileLoggerManager = try FileLoggerManager(
+            fileManager: fileManager,
+            userDefaults: userDefaults,
             dateFormatter: DateFormatter.monthsDaysTimeFormatter,
+            externalLogger: { print($0) },
```

Comment:
Nedava smysl aby byli všechny parametry propagovány z venku? Celkově bych nejradši zrušil FileLogger, protože prakticky jediné co dela FileLogger je delegovani metod na FileLoggerManager a tvori tak zbytečnou mezivrstvu bez přidané hodnoty.

Reply:
jj, nedavalo to moc smysl. Zmergoval jsem a uz existuje jen FileLogger.